src/Stratis.Bitcoin/Configuration/TextFileConfiguration.cs

```diff
@@ -137,15 +141,21 @@ public string[] GetAll(string key)
         /// <typeparam name="T">Type of the argument value.</typeparam>
         /// <param name="key">Name of the argument.</param>
         /// <param name="defaultValue">Default value to return if no argument value is defined.</param>
+        /// <param name="logger">The settings logger to use to log the value. Logs on Debug level.</param>
         /// <returns>Value of the argument or a default value if no value was set.</returns>
-        public T GetOrDefault<T>(string key, T defaultValue)
+        public T GetOrDefault<T>(string key, T defaultValue, ILogger logger = null)
         {
             if (!this.args.TryGetValue($"-{key}", out List<string> values))
+            {
+                logger?.LogDebug("Default value '{0}' was selected for the key '{1}'.", defaultValue, key);
                 return defaultValue;
+            }
 
             try
             {
-                return this.ConvertValue<T>(values[0]);
+                var value = this.ConvertValue<T>(values[0]);
```

Comment:
I wonder if we should change the behaviour and throw in case `values.count >1`

Reply:
There could be two values if the same setting is supplied on the command line as in the configuration file. Since we always add the command line first, taking the first value ensures that the command line takes preference. So, no, we don't want to throw as it would be normal to sometimes want to override the configuration file setting.

Reply:
Yeah, it's a target behavior. In case config have some setting we should always be able to override from command line args since they are more prioritized.

Reply:
ah yes correct thanks.